llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2840,6 +2875,134 @@ class VPBranchOnMaskRecipe : public VPRecipeBase {
   }
 };
 
+class VPConditionalScalarAssignmentHeaderPHIRecipe final
+    : public VPHeaderPHIRecipe {
+public:
+  VPConditionalScalarAssignmentHeaderPHIRecipe(PHINode *Phi,
+                                               VPValue *VPInitData)
+      : VPHeaderPHIRecipe(VPDef::VPConditionalScalarAssignmentHeaderPHISC, Phi,
+                          VPInitData) {}
+
+  ~VPConditionalScalarAssignmentHeaderPHIRecipe() override = default;
+
+  VPConditionalScalarAssignmentHeaderPHIRecipe *clone() override {
+    return new VPConditionalScalarAssignmentHeaderPHIRecipe(
+        cast<PHINode>(getUnderlyingInstr()), getOperand(0));
+  }
+
+  void execute(VPTransformState &State) override;
+
+  InstructionCost computeCost(ElementCount VF,
+                              VPCostContext &Ctx) const override;
+
+#if !defined(NDEBUG) || defined(LLVM_ENABLE_DUMP)
+  /// Print the recipe.
+  void print(raw_ostream &O, const Twine &Indent,
+             VPSlotTracker &SlotTracker) const override;
+#endif
```

Comment:
I think dump() functions, which are more useful within lldb, are missing?

Reply:
No other recipes in this file have dump functions.

Reply:
No other recipes seem to have a dump function. Am I missing something?

Reply:
There's VPDef::dump() and VPValue::dump() that should automatically do the right thing with these print functions. No change necessary.